apps/framework-cli/src/infrastructure/olap/clickhouse/config.rs

```diff
@@ -69,23 +69,34 @@ impl Default for ClickHouseConfig {
 /// Supports multiple URL schemes (https, clickhouse) and extracts database name from path or query parameter.
 /// Automatically determines SSL usage based on scheme and port.
 pub fn parse_clickhouse_connection_string(conn_str: &str) -> anyhow::Result<ClickHouseConfig> {
-    let url = convert_http_to_clickhouse(conn_str)?;
+    let url = Url::parse(conn_str)?;
 
     let user = url.username().to_string();
     let password = url.password().unwrap_or("").to_string();
     let host = url.host_str().unwrap_or("localhost").to_string();
 
+    let mut http_port: Option<u16> = None;
+    let mut native_port: Option<u16> = None;
+
     // Determine SSL based on scheme and port
     let use_ssl = match url.scheme() {
-        "https" => true,
-        "clickhouse" => url.port().unwrap_or(9000) == 9440,
+        "https" => {
+            http_port = Some(url.port().unwrap_or(443));
+            true
+        }
+        "clickhouse" => {
+            let port = url.port().unwrap_or(9000);
+            native_port = Some(port);
+            port == 9440
+        }
         _ => url.port().unwrap_or(9000) == 9440,
     };
 
-    let port = url.port().unwrap_or(if use_ssl { 9440 } else { 9000 }) as i32;
+    let http_port = http_port.unwrap_or(if use_ssl { 8443 } else { 8123 }) as i32;
```

Comment:
How do I use this against the local clickhouse? `http://panda:pandapass@localhost:18123/local`

Looks like use_ssl will be false, http_port starts as none so it becomes 8123?